pytest_twisted.py

```diff
@@ -88,18 +92,103 @@ def _pytest_pyfunc_call(pyfuncitem):
 
 
 def pytest_pyfunc_call(pyfuncitem):
-    if gr_twisted is not None:
-        if gr_twisted.dead:
-            raise RuntimeError("twisted reactor has stopped")
+    if _instances.gr_twisted is not None:
+        if _instances.gr_twisted.dead:
+            raise RuntimeError('twisted reactor has stopped')
 
         def in_reactor(d, f, *args):
             return defer.maybeDeferred(f, *args).chainDeferred(d)
 
         d = defer.Deferred()
-        reactor.callLater(0.0, in_reactor, d, _pytest_pyfunc_call, pyfuncitem)
+        _instances.reactor.callLater(
+            0.0, in_reactor, d, _pytest_pyfunc_call, pyfuncitem
+        )
         blockon_default(d)
     else:
-        if not reactor.running:
-            raise RuntimeError("twisted reactor is not running")
-        blockingCallFromThread(reactor, _pytest_pyfunc_call, pyfuncitem)
+        if not _instances.reactor.running:
+            raise RuntimeError('twisted reactor is not running')
+        blockingCallFromThread(
+            _instances.reactor, _pytest_pyfunc_call, pyfuncitem
+        )
     return True
+
+
+@pytest.fixture(scope="session", autouse=True)
+def twisted_greenlet(request, reactor):
+    request.addfinalizer(stop_twisted_greenlet)
+    return _instances.gr_twisted
+
+
+def init_default_reactor():
+    import twisted.internet.default
+
+    module = inspect.getmodule(twisted.internet.default.install)
+
+    module_name = module.__name__.split('.')[-1]
+    reactor_type_name, = (
+        x
+        for x in dir(module)
+        if x.lower() == module_name
+    )
+    reactor_type = getattr(module, reactor_type_name)
+
+    _install_reactor(
```

Comment:
This seems super hacky but I wasn't sure how else to get the type that the default reactor would be without constructing it (or changing this part of twisted, which I think would be good).